service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/MessageJudgeControllerSubmittedTest.java

```diff
@@ -69,10 +78,59 @@ void shouldNotifyJudicialMessageRecipientWhenNewJudicialMessageAdded() throws No
             "sender", "sender@fpla.com",
             "urgency", "High",
             "hasUrgency", "Yes",
-            "note", "Some note"
+            "latestMessage", MESSAGE
         );
 
         verify(notificationClient).sendEmail(
             NEW_JUDICIAL_MESSAGE_ADDED_TEMPLATE, JUDICIAL_MESSAGE_RECIPIENT, expectedData, "localhost/12345");
     }
+
+    @Test
+    void shouldNotifyJudicialMessageRecipientWhenNewJudicialMessageReplyAdded() throws NotificationClientException {
+        JudicialMessage latestJudicialMessage = JudicialMessage.builder()
+            .recipient(JUDICIAL_MESSAGE_RECIPIENT)
+            .sender("sender@fpla.com")
+            .urgency("High")
+            .latestMessage(REPLY)
+            .messageHistory(MESSAGE + "/n" + REPLY)
+            .build();
+
+        DynamicList dynamicList = DynamicList.builder()
+            .value(DynamicListElement.builder()
+                .code(SELECTED_DYNAMIC_LIST_ITEM_ID)
+                .build())
+            .build();
+
+        CaseDetails caseDetails = CaseDetails.builder()
```

Comment:
Can use case data here. If you set the id of case data to be `CASE_REFERENCE` then when you call `asCaseDetails` it will automatically put it in the correct field

Reply:
And by using CaseData you would get the type safety that the Map doesn't provide